server/server.js

```diff
@@ -1,481 +1,106 @@
-            comments:[
-                { author:"Fran", content:"Nope"},
-                { author:"Greg", content:"Noooooo" },
-                { author:"Hank", content:"Yes please"},
-                { author:"Ian", content:"Nah"}]
-            },
-        {
-                title:"too To TWO",
-                id:4,
-                author: "Jill",
-                comments:[
-                    { author:"Kim", content:"two"},
-                    { author:"Lisa", content:"to" },
-                    { author:"Mary", content:"too"}]
-            
-        }]
-
-    
-        console.log("someone sent a request home");
-        response.send(db.comments);
-    });
-    
-    // Array with comments
-    var threads = 
-        [{ title:"this or that",
-            id:1,
-            author: "Aaron",
-            content: "thread content",
-            comments:[
-                { author:"Brad", content:"this"},
-                { author:"Clark", content:"that" },
-                { author:"Dave", content:"that too"}]
-            },
-            {
-                title:"Yea or nah",
-                id:2,
-                author: "Eli",
-                content: "thread content",
-                comments:[
-                    { author:"Fran", content:"Nope"},
-                    { author:"Greg", content:"Noooooo" },
-                    { author:"Hank", content:"Yes please"},
-                    { author:"Ian", content:"Nah"}]
-            },
-            {
-            title:"Yea or nah",
-            id:3,
-            author: "Eli",
-            content: "thread content",
-            comments:[
-                { author:"Fran", content:"Nope"},
-                { author:"Greg", content:"Noooooo" },
-                { author:"Hank", content:"Yes please"},
-                { author:"Ian", content:"Nah"}]
-            },
-        {
-                title:"too To TWO",
-                id:4,
-                author: "Jill",
-                comments:[
-                    { author:"Kim", content:"two"},
-                    { author:"Lisa", content:"to" },
-                    { author:"Mary", content:"too"}]
-            
-        }]
-
-    
-    
-    app.listen(3000,function(){
-        console.log("listening in port 3000");
-    });
\ No newline at end of file
+const es6Renderer = require("express-es6-template-engine");
+app.engine("html", es6Renderer);
+app.set("views", "templates");
+app.set("view engine", "html");
+
+app.get("/api/threads", function (request, response, next) {
+  console.log("someone sent a request home");
+  response.json(db);
+});
+
+// post to /api/threads is a different route than get to /api/threads
+app.post("/api/threads", function (request, response, next) {
+  console.log("someone sent a request home");
+  response.json("Post add to forum");
+});
+
+// the :id param will be used to identify the thread we want to post a comment to
+app.post("/api/threads/:id/comment", function (request, response, next) {
```

Comment:
This is the handler for the endpoint that accepts a comment for the thread identified by :id

- look up the index of the thread having id = `request.params.id` (the :id in the route). We get the index because we will use it to identify the thread we wish to push the comment onto. If we just used the `find` func, we would get the value...but what we really need is a reference to the object in the array so we can mutate that object.
- push the new comment onto the array of comments in the thread at index position `idx`. We use the `author` and `content` properties of the request body for these values.